src/GeniusYield/Types/PubKeyHash.hs

```diff
@@ -43,14 +44,18 @@ newtype GYPubKeyHash = GYPubKeyHash (Api.Hash Api.PaymentKey)
     deriving stock Show
     deriving newtype (Eq, Ord, IsString)
 
-class CanSignTx a where
+class AsPubKeyHash a where
```

Comment:
A new class `AsPubKeyHash` is introduced with essentially same members as earlier `CanSignTx`. Idea is to make `AsPubKeyHash` a super class of `CanSignTx` since `AsPubKeyHash` is useful for other credentials which do not have any relation with signing such as `GYStakePoolId`.